### ISSUE TITLE
Wasm compatibility of web socket

### ISSUE THREAD

**Issue by gedw99** [anycable/anycable-go#168]
If you changed from using gorilla to one of the wasm compatible web socket libs then people could use wasm plugins with your system.

There’s a few one out there that work with nats. Just ask if you’re curious.

then peopke can just write in any language and compile to wasm and tap into the nats stream over web sockets.

I use , python, typescript and golang .  Python for ML , rust for perf etc . 

Basically it makes things agnostic to the language .

**Comment by gedw99** [comment]
https://pkg.go.dev/nhooyr.io/websocket Works with wasm .

**Comment by gedw99** [comment]
Here is a nice analysis of scaling it out to a million concurrent users 

https://centrifugal.github.io/centrifugo/blog/scaling_websocket/

**Comment by gedw99** [comment]
On a side note quic can also work well with nats .

multipathing so that when uses WAN changes it auto adapts is an awesome feature over web sockets.

seen many nats / quic systems work this way on the gateway level.
